sdk/attestation/attestation/src/constants.ts

```diff
@@ -1,4 +1,4 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-export const SDK_VERSION: string = "1.0.0-beta.5";
+export const SDK_VERSION: string = "1.0.0";
```

Comment:
we should get rid of this, where is it used?

Reply:
It's imported into the clients and is used as:
```
   // The below code helps us set a proper User-Agent header on all requests
    const libInfo = `azsdk-js-api-security-attestation/${SDK_VERSION}`;
    if (!options.userAgentOptions) {
      options.userAgentOptions = {};
    }
    if (options.userAgentOptions.userAgentPrefix) {
      options.userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
    } else {
      options.userAgentOptions.userAgentPrefix = libInfo;
    }
```

Without that, we use the generated code's user agent logic, which claims that the version is 1.0.0-beta.2 (which comes from readme.md - apparently prepareRelease doesn't update this version number.

Reply:
And the tables SDK has essentially the exact same construct if it matters.

Reply:
Ok, I investigated more and I found that agent header has a duplicate in it:
`"user-agent":"azsdk-js-api-security-attestation/1.0.0 azsdk-js-attestation/1.0.0 core-rest-pipeline/1.1.2 Node/v14.17.0 OS/(x64-Linux-5.10.16.3-microsoft-standard-WSL2)"`

This duplication is because code gen became smarter and can generate this handling code for us, see https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/attestation/attestation/src/generated/generatedClientContext.ts#L34-L38

So to fix this, it is safe to get rid of the `SDK_VERSION` constant in `src/constants.ts` and any code that use it (I think it is mentioned in `package.json` too).

Reply:
I'm open to doing this, but the prepareRelease script doesn't update the readme.md file for new updates. That means that we're going to have to manually update the readme.md on every release, which seems EXTREMELY fragile - the tooling knows how to update the constants.ts file and the changelog.md file, but it doesn't understand how to update any of the other files.

So I'd rather keep the code that works instead of somethign that requires manual intervention.

Reply:
Aaah - I think I figured out hwo to make this work. It actually has to be in the package.json file.

Reply:
Digging into this - most packages have an autoupdate of their swagger.md file, but they all still have the constants.ts file and use it essentially the same way as attestation does.

And updating the readme.md isn't sufficient because the swagger isn't regenerated regularly. So I think the constants.ts file is needed (I can't find any examples of SDKs which don't do it this way :().

Reply:
Well we're talking about two separate but related issues:

>  the prepareRelease script doesn't update the readme.md file for new updates.

As you figured, you can make the script update it by doing this: https://github.com/Azure/azure-sdk-for-js/blob/ab936f586aa7082260b7472025ccea3a97e7114a/sdk/schemaregistry/schema-registry/package.json#L48-L51

> but they all still have the constants.ts file and use it essentially the same way as attestation does.

And they have the same issue! Why do you want to have a user-agent header with duplicate versions or even worse with contradictory ones? btw Schema Registry got it right here: https://github.com/Azure/azure-sdk-for-js/blob/ab936f586aa7082260b7472025ccea3a97e7114a/sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts#L44-L67

> And updating the readme.md isn't sufficient because the swagger isn't regenerated regularly

This is a valid concern, I think we should address this by factoring out the version string to its own variable in https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/attestation/attestation/src/generated/generatedClientContext.ts#L34-L38 and add that file to the constantsPath in `package.json`.

I created this issue to fix other packages: https://github.com/Azure/azure-sdk-for-js/issues/16623

Reply:
I created this issue to update the code generator: https://github.com/Azure/autorest.typescript/issues/1125

Reply:
I created a fix for attestation here: https://github.com/Azure/azure-sdk-for-js/pull/16625